scripts/generate_script_functions.py

```diff
@@ -0,0 +1,75 @@
+import sys
+import re
+
+opcode_names = {};
+unimplemented_names = {};
+module_files = sys.argv[1:]
+scmfile = "SCM.ini"
+methods = []
+
+header_template = """/*
+\tTo implement an opcode, add a function to the correct module file and update
+\tthe bindUnimplemented to bindFunction().
+*/
+#define bindUnimplemented(opcode, args, func) \\
+\tbind(opcode, [](const ScriptArguments&) { return false; }, args, #func)
+"""
+
+
+
+# import each existing module file
+for module in module_files:
+    with open(module) as f:
+        for line in f:
+            m = re.search("bindFunction\(\s?0x([0-9A-Fa-f]+), ([^0-9][^,]+),", line)
+            if m is not None:
+                opcode = int(m.group(1), 16)
+                name = m.group(2)
+                opcode_names[opcode] = name
+            m = re.search("bindUnimplemented\(\s?0x([0-9A-Fa-f]+), ([^0-9][^,]+),", line)
+            if m is not None:
+                opcode = int(m.group(1), 16)
+                name = m.group(2)
+                unimplemented_names[opcode] = name
+            m = re.search("bindFunction\(\s?0x([0-9A-Fa-f]+), (-?\d+), ([0-9A-Za-z_<>]+)", line)
+            if m is not None:
+                opcode = int(m.group(1), 16)
+                name = m.group(3)
+                opcode_names[opcode] = name
+            m = re.search("bindUnimplemented\(\s?0x([0-9A-Fa-f]+), (-?\d+), ([0-9A-Za-z_<>]+)", line)
+            if m is not None:
+                opcode = int(m.group(1), 16)
+                name = m.group(3)
+                unimplemented_names[opcode] = name
```

Comment:
Your current solution works for now but I think in the long term searching for implemented opcodes should be done differently or.. not at all.

Given that larger SCM.ini changes are highly unlikely in the future we could generate all function names from that.
We'd then generate all binds (there wouldn't be anything unimplemented) and stubs for all functions (with type from SCM.ini).
If we want to bind functions to templates we'd do it near the implementation instead of in the bind.

We can then simply turn stubs into functions as we see fit.
In the *highly* unlikely event that there is a major rename / refactoring in the SCM.ini we'd just generate a short-script to search for the stubs which need a rename.

Having all stubs ready for implementation also reduces the burden to actually implement a feature. In the long run this will save time. (0x300 opcodes * optimistic 10 seconds to change from bindUnimplemented to a bindFunction and adding a stub with argument parsing:  7680 seconds ~> 2 hours)

This means, that as a one-time-solution we need a tool to rename our existing script functions into what SCM.ini would expect AND we need a handful of rules to rename opcodes (such as "/=" becoming something like: "divide_by" etc.)
We should also have pseudo data-types such as LocalInteger and GlobalInteger we can use as arguments to the stub.
Information which is not yet present in SCM.ini should be added by us, possibly in the form as an overlay SCM.ini, patch or seperate file. That could contain information that a certain argument is a local/global/garage/crane/certain type of pickup/.. .
We could, however, also just cast in the actual implementation functions.